packages/@ama-sdk/core/src/plugins/session-id/session-id.request.ts

```diff
@@ -34,22 +35,28 @@ export class SessionIdRequest implements RequestPlugin {
    */
   public sessionId: string;
 
+  /**
+   * Logger (optional, fallback to console logger if undefined)
+   * @protected
+   */
+  protected logger?: Logger;
+
   /**
    * Constructor.
    * @param sessionIdHeader The request header in which the ID will be added.
    * @param activateRequestId Indicates if the request ID part should be added to the ID.
+   * @param logger Logger (optional, fallback to console logger if undefined)
    */
-  constructor(sessionIdHeader = 'Ama-Client-Ref', activateRequestId = true) {
+  constructor(sessionIdHeader = 'Ama-Client-Ref', activateRequestId = true, logger?: Logger) {
```

Comment:
Is there a reason why you declare it as a property and not simply scope it in the constructor?
```suggestion
  constructor(sessionIdHeader = 'Ama-Client-Ref', activateRequestId = true, protected logger?: Logger) {
```

Reply:
this parameter will be removed with the modifications suggested below